nft-example/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+#![no_std]
+#![feature(const_btree_new)]
+
+use codec::{Decode, Encode};
+use gstd::{debug, exec, msg, prelude::*, ActorId};
+use primitive_types::{H256, U256};
+use scale_info::TypeInfo;
+
+pub mod payloads;
+pub use payloads::{ApproveForAllInput, ApproveInput, InitConfig, TransferInput};
+
+pub mod state;
+pub use state::{State, StateReply};
+
+use non_fungible_token::base::NonFungibleTokenBase;
+use non_fungible_token::{Approve, ApproveForAll, NonFungibleToken, Transfer};
```

Comment:
Not sure why `Approve`, `ApproveForAll` and `Transfer` are imported here. Those are just used as action names below, am I wrong?

Reply:
No, these are not the name of actions, these are the names of structs in enum `Events`. `Transfer` struct is necessary since it is used in `mint` and `burn`, but `Approve` and `ApproveForAll` are really redundant, thanks

Reply:
`Approve`, `ApproveForAll` are also necessary, these are structs of message replies, and must be defined in `gstd::metadata` (enum `Events`)